core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -602,9 +667,29 @@ private ExprValue exprAddDateInterval(ExprValue date, ExprValue expr) {
    * @return Date/Datetime resulted from days added to date.
    */
   private ExprValue exprAddDateDays(ExprValue date, ExprValue days) {
-    ExprValue exprValue = new ExprDatetimeValue(date.datetimeValue().plusDays(days.longValue()));
-    return (exprValue.timeValue().toSecondOfDay() == 0 ? new ExprDateValue(exprValue.dateValue())
-        : exprValue);
+    return exprDateApplyDays(date, days.longValue(), true);
+  }
+
+  /**
+   * Adds or subtracts `days` to/from `date`.
+   *
+   * @param date A Date/Time/Datetime/Timestamp value to change.
+   * @param days A days amount to add or subtract.
+   * @param add A flag: true to add, false to subtract.
+   * @return Datetime calculated.
+   */
+  private ExprValue exprDateApplyDays(ExprValue date, Long days, Boolean add) {
```

Comment:
if `date` isn't exclusively a date type, please rename.  `initialDateTime` or `initialValue`? You might want to rename the function too... maybe `exprDateApplyDays
Also please rename add to be a question, such as `isAddition` or `isAdd` or `isPositive`.